packages/editor/src/lib/editor/tools/StateNode.ts

```diff
@@ -159,6 +174,10 @@ export abstract class StateNode implements Partial<TLEventHandlers> {
 
 	// todo: move this logic into transition
 	enter = (info: any, from: string) => {
+		if (debugFlags.measurePerformance.get() && STATE_NODES_TO_MEASURE.includes(this.id)) {
+			this.fpsTracker.start(this.id)
```

Comment:
doesn't have to be for this PR but for you to think about:
it'd be interesting to consider things like, not just FPS, but say in the future we want to track something else like memory or some other metric like raw time (instead of FPS)
i guess what i'm saying is maybe we have a `perfTracker` (instead of only an `fpsTracker`) that does FPS, but also might do other things as well.

Reply:
Yeah, makes sense. Even with this one it's easy to add duration and frame count to the printout, but we might improve it in the future.

It could even track perf for a random subset of prod users.